module/data/item/templates/class.mjs

```diff
@@ -165,12 +165,31 @@ export default class ClassTemplate extends ItemDataModel.mixin(
       nextLevelData.abilities.special.map( ability => fromUuid( ability ) )
     );
 
+
+
     const effects = Array.from( nextLevelData.effects );
 
     await this.parentActor.createEmbeddedDocuments( "Item", [ ...freeAbilityData, ...specialAbilityData ] );
     await this.parentActor.createEmbeddedDocuments( "ActiveEffect", effects );
     // TODO: activate permanent effects immediately
 
+    // increase resource step of the discipline
+
+    const highestClass = this.parentActor.items.filter(
```

Comment:
This seems like a good candidate for a dynamic getter (`get highestClass()`) in the document class itself

Reply:
good idea, did it